src/charm.py

```diff
@@ -95,6 +98,40 @@ def __init__(self, *args):
             self.fiveg_n3_provider.on.fiveg_n3_request, self._on_fiveg_n3_request
         )
 
+    def _is_pod_restart_required(self) -> None:
+        """Checks if existing NAD config is different from the required ones which are set in the configuration  # noqa: E501, W505.
+
+        If there is difference, sets the _restart_pod attribute to True.
+        """
+        if (
+            self._get_nad_active_mtu_size(ACCESS_NETWORK_ATTACHMENT_DEFINITION_NAME)
+            != self._get_access_interface_mtu_config()  # noqa: W503
+        ):
+            self._restart_pod = True
```

Comment:
I'm not a huge fan of setting attributes here. The problem with charms is that it's easy to get confused. It's possible to set the value on 1 event and this value is not accessible on follow up events. I rather us call this method directly every time we need it.

In other words, let's have this method return a bool instead of setting "restart_pod".

Reply:
Done